### PR TITLE
feat: add agentcore destroy command

### PR DESCRIPTION
**Pull request by huanshenyi**
## Description

Brief description of changes

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Testing

- [x] Unit tests pass locally
- [x] Integration tests pass (if applicable)
- [x] Test coverage remains above 80%
- [x] Manual testing completed

## Checklist

 - [x] My code follows the project's style guidelines (ruff/pre-commit)
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] Any dependent changes have been merged and published

## Security Checklist

- [x] No hardcoded secrets or credentials
- [x] No new security warnings from bandit
- [x] Dependencies are from trusted sources
- [x] No sensitive data logged

  ## Breaking Changes

  N/A - This is an additive feature that doesn't modify existing functionality.

  ## Additional Notes

  ### New Command Usage
  ```bash
  agentcore destroy                    # Destroy default agent (with confirmation)
  agentcore destroy --agent my-agent  # Destroy specific agent
  agentcore destroy --dry-run         # Preview what would be destroyed
  agentcore destroy --force           # Skip confirmation prompts

  Resources Cleaned Up

  - Bedrock AgentCore endpoints
  - Bedrock AgentCore agent runtimes
  - ECR images (latest tag only for safety)
  - CodeBuild projects
  - IAM execution roles (only if not used by other agents)
  - Agent deployment configuration

  Safety Features

  - Confirmation prompts by default
  - Dry-run mode to preview changes
  - Shared resource protection (won't delete IAM roles used by multiple agents)
  - Comprehensive error handling and logging
  - Graceful handling of already-deleted resources

  Test Coverage

  - 25+ comprehensive unit tests covering all scenarios
  - CLI integration tests
  - Error handling and edge case testing
  - Mock-based testing for AWS service interactions

### CODE REVIEW

File: src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+        if dry_run:
+            result.resources_removed.append(f"AgentCore endpoint for agent {agent_id} (DRY RUN)")
+            return
+
+        # Try to get and delete endpoint
+        try:
+            endpoint_response = client.get_agent_runtime_endpoint(agent_id)
+            endpoint_arn = endpoint_response.get("agentRuntimeEndpointArn")
+            if endpoint_arn:
+                # Note: There's no direct delete endpoint method in the current client
```
Comment (Vivekbhadauria1) [inline src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py]: we should use https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-agentcore-control/client/delete_agent_runtime_endpoint.html.

**Review by Vivekbhadauria1 — CHANGES_REQUESTED** [review]
Thanks for merging latest changes. Needs a couple of changes -  
1. Code build IAM role starting with prefix AmazonBedrockAgentCoreSDKCodeBuild* needs to be deleted if developers uses code-build.
2. The delete endpoint needs to special case "DEFAULT" endpoint that can not be explicitly created/deleted.

File: src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+        result.warnings.append(f"Error during CodeBuild cleanup: {e}")
+        log.warning("Error during CodeBuild cleanup: %s", e)
+
+def _destroy_codebuild_iam_role(
+    session: boto3.Session,
+    agent_config: BedrockAgentCoreAgentSchema,
+    result: DestroyResult,
+    dry_run: bool,
+) -> None:
+    """Remove all CodeBuild IAM roles with prefix AmazonBedrockAgentCoreSDKCodeBuild*"""
```
Comment (Vivekbhadauria1) [inline src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py]: only delete the code build role associated with agent.  You can read the config file similar to agent's execution role. The code build role is at "codebuild.execution_role" in the agent's config.

File: src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+        return
+        
+    try:
+        # Note: IAM is a global service, but we specify region for consistency
+        iam_client = session.client("iam", region_name=agent_config.aws.region)
+        role_arn = agent_config.codebuild.execution_role
+        role_name = role_arn.split("/")[-1]
+        
+        if dry_run:
+            result.resources_removed.append(f"DRY RUN: would remove CodeBuild IAM role: {role_name}")
```
Comment (Vivekbhadauria1) [inline src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py]: Inconsistent with other dry logs. should be "CodeBuild IAM role <Rolename> (DRY RUN) "

File: src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+    config_path: Path,
+    project_config: BedrockAgentCoreConfigSchema,
+    agent_name: str,
+    result: DestroyResult,
+) -> None:
+    """Remove agent configuration from the config file."""
+    try:
+        # Clear the bedrock_agentcore deployment info but keep the agent config
+        if agent_name in project_config.agents:
+            agent_config = project_config.agents[agent_name]
```
Comment (Vivekbhadauria1) [inline src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py]: The agent specific entry should be removed and not just the bedrock_agentcore. This causes the subsequent status/configure/launch calls to fail.

File: src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+def _cleanup_agent_config(
+    config_path: Path,
+    project_config: BedrockAgentCoreConfigSchema,
+    agent_name: str,
+    result: DestroyResult,
+) -> None:
+    """Remove agent configuration from the config file."""
+    try:
+        # Clear the bedrock_agentcore deployment info but keep the agent config
+        if agent_name in project_config.agents:
```
Comment (Vivekbhadauria1) [inline src/bedrock_agentcore_starter_toolkit/operations/runtime/destroy.py]: Default agent name needs to be cleaned up in the config. If there is only one agent and its being destroyed we can remove the config file. Else, we can set the default to next valid agent in the file.

**Review by Vivekbhadauria1 — CHANGES_REQUESTED** [review]
Thanks! left a few comment on config file cleanup that is being left in an invalid state post destroy.

File: README.md [hunk 6]
```diff
@@ -40,6 +40,36 @@ AgentCore Runtime is a secure, serverless runtime purpose-built for deploying an
...
+# Configure your agent for deployment
+agentcore configure --entrypoint my_agent.py --name my-production-agent
+
+# Check deployment status
+agentcore status
+
+# Invoke your deployed agent
+agentcore invoke '{"prompt": "Hello from Bedrock AgentCore!"}'
+```
+
```
Comment (Vivekbhadauria1) [inline README.md]: This has been moved to AgentCore Runtime Quick start and need not be present at top-level documentation. Please update bedrock-agentcore-starter-toolkit/user-guide/runtime/quickstart instead.

File: README.md [hunk 7]
```diff
@@ -73,6 +102,27 @@ AgentCore Import-Agent enables seamless migration of existing Amazon Bedrock Age
...
+- [Utility Overview](https://github.com/aws/bedrock-agentcore-starter-toolkit/blob/main/documentation/docs/user-guide/import-agent/overview.md)
+
+## 📚 About Amazon Bedrock AgentCore
+
+Amazon Bedrock AgentCore enables you to deploy and operate highly effective agents securely, at scale using any framework and model. With AgentCore, developers can accelerate AI agents into production with enterprise-grade scale, reliability, and security. The platform provides:
+
+- **Composable Services**: Mix and match services to fit your needs
+- **Framework Flexibility**: Works with Strands, LangGraph, CrewAI, Strands, and more
+- **Any Model Support**: Not locked into specific models
+- **Enterprise Security**: Built-in identity, isolation, and access controls
```
Comment (Vivekbhadauria1) [inline README.md]: Please sync content from https://github.com/aws/bedrock-agentcore-starter-toolkit. Ideally there should be no new change to this file for destroy.

**Review by Vivekbhadauria1 — CHANGES_REQUESTED** [review]
A couple of doc related comments.

File: README.md (hunk 7 above)
Comment (huanshenyi) [reply README.md]: @Vivekbhadauria1 
Thank you so much for your patience and guidance! I'm really grateful I could help you address the documentation feedback. Your
  implementation of the destroy functionality is excellent work.

  We're still working through the review process, but I appreciate you taking the time to collaborate on getting everything just right. It's
   been a pleasure working with you on this contribution! 🙏

**Review by Vivekbhadauria1 — APPROVED** [review]

**Review by Vivekbhadauria1 — CHANGES_REQUESTED** [review]
Hey, The tests are failing - https://github.com/aws/bedrock-agentcore-starter-toolkit/actions/runs/17495999286/job/49711013016?pr=100. Please take a look.

**Review by Vivekbhadauria1 — APPROVED** [review]
